bot/exts/evergreen/xkcd.py

```diff
@@ -0,0 +1,90 @@
+import logging
+import re
+from random import randint
+from typing import Dict, Optional, Union
+
+from discord import Embed
+from discord.ext import tasks
+from discord.ext.commands import Cog, Context, command
+
+from bot.bot import Bot
+from bot.constants import Colours
+
+log = logging.getLogger(__name__)
+
+COMIC_FORMAT = re.compile(r"latest|[0-9]+")
+BASE_URL = "https://xkcd.com"
+
+
+class XKCD(Cog):
+    """Retrieving XKCD comics."""
+
+    def __init__(self, bot: Bot) -> None:
+        self.bot = bot
+        self.latest_comic_info: Dict[str, Union[str, int]] = {}
+        self.get_latest_comic_info.start()
+
+    def cog_unload(self) -> None:
+        """Cancels refreshing of the task for refreshing the most recent comic info."""
+        self.get_latest_comic_info.cancel()
+
+    @tasks.loop(minutes=30)
+    async def get_latest_comic_info(self) -> None:
+        """Refreshes latest comic's information ever 30 minutes. Also used for finding a random comic."""
+        async with self.bot.http_session.get(f"{BASE_URL}/info.0.json") as resp:
+            if resp.status == 200:
+                self.latest_comic_info = await resp.json()
+            else:
+                log.debug(f"Failed to get latest XKCD comic information. Status code {resp.status}")
+
+    @command(name="xkcd")
+    async def fetch_xkcd_comics(self, ctx: Context, comic: Optional[str]) -> None:
```

Comment:
```suggestion
    async def fetch_xkcd_comics(self, ctx: Context, comic: Optional[Union[int, str]]) -> None:
```
This is just me being pointlessly nitpicky, especially since the comic number needs to be `str` to include in the url (though f-string will handle that.)
Feel free to ignore this comment. Everything else LGTM.

Reply:
The regex on [line 51](https://github.com/python-discord/sir-lancebot/pull/415/files#diff-212b6fe76d52cce2157539797e4d693dec1aa5ffc5aaefd5b01056ac54de837fR51) will also need `comic` to be an `str` so I guess we could leave this as it is.

Reply:
I actually had this in my code before, but decided against it. 

There's no reason for the `comic` parameter to explicitly be an integer.

Reply:
alright, cheers :+1: